From 3284935177b3698b3ce7d5086d86b761bbd1bb95 Mon Sep 17 00:00:00 2001
From: Tamizh Chelvam Raja <quic_tamizhr@quicinc.com>
Date: Wed, 20 Dec 2023 11:45:07 +0530
Subject: [PATCH] wifi: ath12k: Remove unused setup configuration to improve
 memory headroom

Currently memory allocation for multiple hardware rings
done in host as these memory will be used by FW/HW.
But in case of tcl_cmd_ring and tcl_status_ring neither
FW nor HW uses host allocated memory and it uses memory
allocated by FW itself. So avoid tcl_cmd_ring and tcl_status_ring
setup in host. And reduce DP_REO_DST_RING_MAX to 4 as driver uses
only 4 REO_DST ring.

Signed-off-by: Tamizh Chelvam Raja <quic_tamizhr@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.c | 19 -------------------
 drivers/net/wireless/ath/ath12k/dp.h |  4 +---
 2 files changed, 1 insertion(+), 22 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp.c b/drivers/net/wireless/ath/ath12k/dp.c
index 1520049..f9bd565 100644
--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -827,7 +827,6 @@ void ath12k_dp_srng_hw_ring_disable(struct ath12k_base *ab)
 	for (i = 0; i < DP_REO_DST_RING_MAX; i++)
 		ath12k_dp_srng_hw_disable(ab, &dp->reo_dst_ring[i]);
 	ath12k_dp_srng_hw_disable(ab, &dp->wbm_desc_rel_ring);
-	ath12k_dp_srng_hw_disable(ab, &dp->tcl_cmd_ring);
 
 	for(i = 0; i < ab->hw_params->max_tx_ring; i++) {
 		ath12k_dp_srng_hw_disable(ab, &dp->tx_ring[i].tcl_data_ring);
@@ -855,8 +854,6 @@ static void ath12k_dp_srng_common_cleanup(struct ath12k_base *ab)
 		ath12k_dp_srng_cleanup(ab, &dp->tx_ring[i].tcl_comp_ring);
 		ath12k_dp_srng_cleanup(ab, &dp->tx_ring[i].tcl_data_ring);
 	}
-	ath12k_dp_srng_cleanup(ab, &dp->tcl_status_ring);
-	ath12k_dp_srng_cleanup(ab, &dp->tcl_cmd_ring);
 	ath12k_dp_srng_cleanup(ab, &dp->wbm_desc_rel_ring);
 
 #ifdef CPTCFG_ATH12K_PPE_DS_SUPPORT
@@ -882,22 +879,6 @@ static int ath12k_dp_srng_common_setup(struct ath12k_base *ab)
 		goto err;
 	}
 
-	ret = ath12k_dp_srng_setup(ab, &dp->tcl_cmd_ring, HAL_TCL_CMD, 0, 0,
-				   DP_TCL_CMD_RING_SIZE);
-	if (ret) {
-		ath12k_warn(ab, "failed to set up tcl_cmd ring :%d\n", ret);
-		goto err;
-	}
-
-	if (!ath12k_dp_umac_reset_in_progress(ab)) {
-		ret = ath12k_dp_srng_setup(ab, &dp->tcl_status_ring, HAL_TCL_STATUS,
-					   0, 0, DP_TCL_STATUS_RING_SIZE);
-		if (ret) {
-			ath12k_warn(ab, "failed to set up tcl_status ring :%d\n", ret);
-			goto err;
-		}
-	}
-
 	for (i = 0; i < ab->hw_params->max_tx_ring; i++) {
 		map = ab->hw_params->hal_ops->tcl_to_wbm_rbm_map;
 		tx_comp_ring_num = map[i].wbm_ring_num;
diff --git a/drivers/net/wireless/ath/ath12k/dp.h b/drivers/net/wireless/ath/ath12k/dp.h
index a313e25..e6ccf33 100644
--- a/drivers/net/wireless/ath/ath12k/dp.h
+++ b/drivers/net/wireless/ath/ath12k/dp.h
@@ -183,7 +183,7 @@ struct ath12k_pdev_dp {
 #define DP_TX_IDR_SIZE			DP_TX_COMP_RING_SIZE
 #define DP_TCL_CMD_RING_SIZE		32
 #define DP_TCL_STATUS_RING_SIZE		32
-#define DP_REO_DST_RING_MAX		8
+#define DP_REO_DST_RING_MAX		4
 #define DP_REO_DST_RING_SIZE		2048
 #define DP_REO_REINJECT_RING_SIZE	32
 #define DP_RX_RELEASE_RING_SIZE		4096
@@ -412,8 +412,6 @@ struct ath12k_dp {
 	u8 idle_link_rbm_id;
 	struct dp_srng wbm_idle_ring;
 	struct dp_srng wbm_desc_rel_ring;
-	struct dp_srng tcl_cmd_ring;
-	struct dp_srng tcl_status_ring;
 	struct dp_srng reo_reinject_ring;
 	struct dp_srng rx_rel_ring;
 	struct dp_srng reo_except_ring;
-- 
2.17.1

